flake.nix

```
@@ -3,5 +3,6 @@
     nixosModules.impermanence = import ./nixos.nix;
     nixosModules.home-manager.impermanence = import ./home-manager.nix;
     nixosModule = self.nixosModules.impermanence;
+    hmModule = self.nixosModules.home-manager.impermanence;
```

I'd use `homeManager` as module name.

---

I think sticking to hmModule would be better to make it consistent with the (admittedly, subjective) rest of the nixos ecosystem.

For instance, home-manager itself officially calls the type of a home-manager module `hmModule`
https://github.com/nix-community/home-manager/blob/ec06f419af79207b33d797064dfb3fc9dbe1df4a/nixos/common.nix#L14

And I've seen other projects in nix-community follow the same naming scheme, for example see
https://github.com/nix-community/nix-doom-emacs/blob/7d7eb4c02eff4228c344d564edeaaf0458d6a742/flake.nix#L126

I don't know whether that naming is ideal, however I think it's okay enough to not change it for consistency's sake.

And if we'd really want to deviate from this informal convention, something like `homeManagerModule` would imo be better suited (because it's a home-manager module being exported, not home-manager itself)

---

Would this be better?
```suggestion
    homeManagerModule = self.homeManagerModules.impermanence;
```
And then change `nixosModules.home-manager.impermanence = import ./home-manager.nix;` to be `homeManagerModules.impermanence = import ./home-manager.nix;`.

---

The idiomatic way to signal a default for through a singular top-level attribute name has been deprecated to instead use `<plural-attr-name>.default`.

So IMO we should use `homeManagerModules.{impermanence,default}`.